### PR TITLE
Save artifacts in cluster claim step

### PR DESCRIPTION
**Pull request by hongkailiu**
/hold
will check the artifact folder in e2e.

### CODE REVIEW

**Comment by hongkailiu** [comment]
seems working.
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_ci-tools/2161/pull-ci-openshift-ci-tools-master-e2e/1413606834840801280/artifacts/e2e/e2e/artifacts/TestMultiStage_e2e-claim/build-resources/

/hold cancel

File: pkg/steps/cluster_claim.go [hunk 1]
```diff
@@ -186,6 +191,11 @@ func getHiveSecret(src *corev1.Secret, name, namespace, testName string) (*corev
 }
 
 func (s *clusterClaimStep) releaseCluster(ctx context.Context, clusterClaim *hivev1.ClusterClaim) error {
+	if err := s.saveArtifacts(ctx, clusterClaim.Namespace, clusterClaim.Name); err != nil {
```
Comment (stevekuznetsov) [inline pkg/steps/cluster_claim.go:194]: nit:

```go
logger := logrus.WithField("clusterClaim.Namespace", clusterClaim.Namespace).WithField("clusterClaim.Name", clusterClaim.Name)
if err := s.saveArtifacts(ctx, clusterClaim.Namespace, clusterClaim.Name, logger); err != nil {
// ...

```

Comment (hongkailiu) [reply pkg/steps/cluster_claim.go:195]: Updated.

File: pkg/steps/cluster_claim.go [hunk 2]
```diff
@@ -202,13 +212,49 @@ func (s *clusterClaimStep) releaseCluster(ctx context.Context, clusterClaim *hiv
...
+		return fmt.Errorf("failed to get the claim %s in namespace %s: %w", name, namespace, err)
+	}
+	data, err := json.MarshalIndent(claim, "", "  ")
+	if err != nil {
+		errs = append(errs, err)
+	}
+	path := filepath.Join(namespaceDir, "claim.json")
+	if err := api.SaveArtifact(s.censor, path, data); err != nil {
+		errs = append(errs, err)
+	}
```
Comment (emilvberglind) [inline pkg/steps/cluster_claim.go:230]: If the above `if err != nil` is true, would this logic ever succeed or is it more efficient to skip over it?

File: pkg/steps/cluster_claim.go [hunk 3]
```diff
@@ -202,13 +212,49 @@ func (s *clusterClaimStep) releaseCluster(ctx context.Context, clusterClaim *hiv
...
+			return fmt.Errorf("failed to get cluster deployment %s in namespace %s: %w", claim.Spec.Namespace, claim.Spec.Namespace, err)
+		}
+		data, err := json.MarshalIndent(clusterDeployment, "", "  ")
+		if err != nil {
+			errs = append(errs, err)
+		}
+		path := filepath.Join(namespaceDir, "clusterDeployment.json")
+		if err := api.SaveArtifact(s.censor, path, data); err != nil {
+			errs = append(errs, err)
+		}
```
Comment (emilvberglind) [inline pkg/steps/cluster_claim.go:244]: Same question here.

File: pkg/steps/cluster_claim.go (hunk 2 above)
Comment (hongkailiu) [reply pkg/steps/cluster_claim.go]: Skipped the `api.SaveArtifact` call  while err on `json.MarshalIndent`.

File: pkg/steps/cluster_claim.go (hunk 3 above)
Comment (hongkailiu) [reply pkg/steps/cluster_claim.go]: Updated here too.

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**